### ISSUE TITLE
 Not localized og tag on FE

### ISSUE THREAD

**Issue by alesvojta** [jonathanheilmann/ext-jh_opengraphprotocol#23]
Hi I'm using the ext on multilingual website and when is the og:title field filled, the value is propagated to other languages and I'm unable to change it.

Example:
Main language is DE. *og:title = Custom OG title*.
Second language is EN, *og:title* is left blank so it should display *EN page title* (is it correct assumption?), but the *DE og:title* is shown. So I'm trying to override it with *EN og:title* but with no luck => again the *DE og:title* is shown.

I'm sorry if I overlooked something.

**Comment by jonathanheilmann** [comment]
This seems to be a buggy behavior.
I'll do some investigations and hopefully present a solution, soon.

**Comment by jonathanheilmann** [comment]
Solving this issue has not been as easy as I hoped. Thanks to @tpinne, you got me on the right track.
It's required to add l10n_mode to TCA to merge if value in translation is blank. While a forced changed mode may break some installations/required output, it's optional for each field and must be set in extension configuration in extension manager.

I would be glad, if someone could test the actual development brach (https://github.com/jonathanheilmann/ext-jh_opengraphprotocol/tree/develop_v1.2.2) and leave some feedback.

**Comment by kraemer-igroup** [comment]
Hi, thanks for the update! I just tested it with an installation with two languages (default + en). The translation now works fine, except on the foreign language always the default image is used.

I guess it should be "$GLOBALS['TSFE']->page['_PAGES_OVERLAY_UID']" instead of "$GLOBALS['TSFE']->id" in this line:
https://github.com/jonathanheilmann/ext-jh_opengraphprotocol/blob/develop_v1.2.2/Classes/Service/OgRendererService.php#L116

I also can not see the extension options? I can click and the configuration icon is displayed in the extension manager but then there are no options...

**Comment by jonathanheilmann** [comment]
Which TYPO3 CMS version do you use?

If you do not set a field in translation (en localization), does TYPO3 CMS render no output for this og-tag or does it use the default language's field?

I've attached two screenshots of the extension configuration. One for version 6.2 and one for 7.6.
Did you try to flush your system cache?
Extension configuration in 6.2:
![Extension configuration in 6.2](https://cloud.githubusercontent.com/assets/2504896/17631827/f8c1fe92-60c5-11e6-90c4-341f0f2a72d3.png)
Extension configuration in 7.6
![Extension configuration in 7.6](https://cloud.githubusercontent.com/assets/2504896/17631828/f8c340d6-60c5-11e6-921b-3bdc2a192af5.png)

Related to images:
I've developed the upper fix in version 7.6.10. In version 6.2.26 this fix does not work and your change has to bee applied.
**Reason:**
6.2 creates this record in database table sys_file_reference:
```
{
	uid: 229,
	pid: 51,
	uid_local: 6,
	uid_foreign: 6,
	tablenames: 'pages_language_overlay',
	fieldname: 'tx_jhopengraphprotocol_ogfalimages'
}
```
Whereas pid is the uid of the pages-record and uid_foreign is the uid of the pages_language_overlay-record.
That's all finde and works as aspected.

7.2.9 and 7.6.10 creates this record in database table sys_file_reference:
```
{
	uid: 229,
	pid: 51,
	uid_local: 6,
	uid_foreign: 51,
	tablenames: 'pages_language_overlay',
	fieldname: 'tx_jhopengraphprotocol_ogfalimages'
}
```
Obviously, version 7.2.x saves the wrong uid_foreign. 51 is the pages' uid and not the pages_language_overlay's uid.

I assume, this is a core bug in 7.2 and maybe related to https://forge.typo3.org/issues/76174 (Patch does not resolve this bug), as the links are not displayed (see screenshots). But I could not find another issue, matching the faulty uid_foreign field.
Links in 6.2:
![Links in 6.2](https://cloud.githubusercontent.com/assets/2504896/17631983/81a67242-60c6-11e6-86a1-1bd053087378.png)
Missing links in 7.6:
![Missing links in 7.6](https://cloud.githubusercontent.com/assets/2504896/17632010/9c5b7b64-60c6-11e6-9b73-39c52d16b1cb.png)

**Comment by jonathanheilmann** [comment]
I've done some more testings for TYPO3 CMS 7.6.x:
You will have to apply the patch from https://forge.typo3.org/issues/76174. Without this, localization of images does not work in TYPO3 CMS 7.6.x.

**Comment by kraemer-igroup** [comment]
Ok, this is strange because I use TYPO3 CMS 7.6.9 and my change was necessary. Maybe this is fixed with the installed filemetadata extension which also "fixes" an issue with FAL and workspaces: https://forge.typo3.org/issues/73076.

Without the change the default image is used.

I did clear all caches with the install tool and could not see the extension options. I even deleted the typo3temp/Cache folder.

**Comment by jonathanheilmann** [comment]
I'll install filemetadata tomorrow and check it.

Really strange: Did you try to deactivate and re-activate the extension? If this dosen't fix it, I'm going to ask the TYPO3 community. I've never had a problem like this before :/

**Comment by jonathanheilmann** [comment]
Now I'm confused: I've reverted the bugfix, removed folder typo3temp/ and cleared all caches.
Actually, all fields in sys_file_reference are set properly and not [as described 5 days ago](https://github.com/jonathanheilmann/ext-jh_opengraphprotocol/issues/23#issuecomment-239516409).

In fact, @kraemer-igroup, your change is the way it should work. If there are issues due to this, it's related to core and not the extension. Activating, deactivating filemetadata doesn't change the behavior.

I've attached a beta of the extension (commit cdb635270d137fa79c88c7ecf02bd98f017f93f9) maybe uploading the extension as zip via Extension Manager, or updating the extension via composer, solves the configuration issue?

[jh_opengraphprotocol_1.2.2-beta_201608171737.zip](https://github.com/jonathanheilmann/ext-jh_opengraphprotocol/files/422952/jh_opengraphprotocol_1.2.2-beta_201608171737.zip)

**Comment by kraemer-igroup** [comment]
Hi, great to hear you solved the translation issue! Also I am relieved TYPO3 did not change this from 6.2 to 7.6 ;-)

I just re-installed the extension and now I can see the configuration... Strange, but your extension seems to work fine and it was just an issue with my installation. Thanks for your effort and suggestions to fix this!

**Comment by jonathanheilmann** [comment]
Issue is resolved and fix is available in version 1.2.2. 
Thanks to all participants!
